### PR TITLE
User hook function for handling unsupported Ethernet frames

### PR DESCRIPTION
**Pull request by evpopov**
Description
-----------
This PR allows users to implement custom Ethernet protocols. Without those changes, users have to modify FreeRTOS+TCP code every time they pull the latest changes. An example usage of the proposed changes is implementing LLDP, PPPoE, or any other Ethernet protocol except for the already supported ones like IP or ARP. More details on the discussion that preceded this PR is available [in the forums](https://forums.freertos.org/t/freertos-tcp-custom-ethernet-protocols/11858/14)

Test Steps
-----------
None

Related Issue
-----------
None

### CODE REVIEW

File: FreeRTOS_IP.c
Comment (htibosch) [inline FreeRTOS_IP.c:207]: Hello @evpopov , thanks for this PR. It extends the possibilities of the stack without adding any real code.
As we already have discussed it on the [FreeRTOS forum](https://forums.freertos.org/t/freertos-tcp-custom-ethernet-protocols), and I think it is good.

**Review by htibosch — APPROVED** [review]
@htibosch approves this PR.

**Review by AniruddhaKanhere — APPROVED** [review]

**Comment by AniruddhaKanhere** [comment]
Hello @evpopov, thank you for the scintillating conversation on the forums and this PR which allows users to handle unsupported frames as they want without modifying the FreeRTOS+TCP stack.
Apologies for the delay with this PR review. I have reviewed and approved it.

We'll work on getting another review and getting this merged!

**Comment by evpopov** [comment]
@AniruddhaKanhere, no worries, take your time.
 I just have a side question: If/when this PR is merged, I can delete my branches from my clone without affecting anything in this repo, right?

**Comment by htibosch** [comment]
Yes, once the branch had got merged, you are free to delete it. GitHub will also inform you about that.
After merging, you are also free delete the entire clone.

I am often keep old branches because they also serve as a personal archive.

**Review by aggarg — APPROVED** [review]

**Pull request merged**
